.vscode/settings.json

```
@@ -9,11 +9,43 @@
   "[html]": {
     "editor.formatOnSave": false
   },
+  "[ejs]": {
+    "editor.formatOnSave": false
+  },
+  "[lua]": {
+    "editor.defaultFormatter": "sumneko.lua",
+    "editor.formatOnSave": false
+  },
```

```suggestion
  "[lua]": {
    "editor.defaultFormatter": "sumneko.lua",
    "editor.formatOnSave": true
  },
```

Should we activate by default for our Lua file ? 

Currently this will be only manual. We have a lot of Lua file, so I opting in could do a lot of style change. 

I don't know what the formatting feature from this extension is worth. 

Other formatter extension on the market
- https://marketplace.visualstudio.com/items?itemName=JohnnyMorganz.stylua
- https://marketplace.visualstudio.com/items?itemName=gea-embedded.metalua-formatter

But probably better to use the other one we already use

---

> Should we activate by default for our Lua file ?

I'd prefer we didn't, to minimize the source control diffs.

---

Fine. Let's manually allow it then.

---

`formatOnSave` can mess a lot with the codebase especially when the codebase is not "clean".

As from the suggestion (on Slack) of using CodeSpell, I would make dedicated PRs to clean the codebase which later could mean you can "format on save" as it will only apply to your currently added code.